packages/flutter/lib/src/material/drawer.dart

```diff
@@ -218,34 +235,72 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
 
   void _move(DragUpdateDetails details) {
     final double delta = details.primaryDelta / _width;
-    switch (Directionality.of(context)) {
-      case TextDirection.rtl:
-        _controller.value -= delta;
-        break;
-      case TextDirection.ltr:
-        _controller.value += delta;
-        break;
+    if (widget.type == DrawerType.START) {
+      if (Scaffold.of(context).isEndSideDrawerOpen() == true) {
+        _controller.value = 0.0;
+      } else {
+        switch (Directionality.of(context)) {
+          case TextDirection.rtl:
+            _controller.value -= delta;
+            break;
+          case TextDirection.ltr:
+            _controller.value += delta;
+            break;
+        }
+      }
+    } else {
+      if (Scaffold.of(context).isDrawerOpen() == true) {
+        _controller.value = 0.0;
+      } else {
+        switch (Directionality.of(context)) {
+          case TextDirection.rtl:
+            _controller.value += delta;
+            break;
+          case TextDirection.ltr:
+            _controller.value -= delta;
+            break;
+        }
+      }      
     }
   }
 
   void _settle(DragEndDetails details) {
-    if (_controller.isDismissed)
-      return;
-    if (details.velocity.pixelsPerSecond.dx.abs() >= _kMinFlingVelocity) {
-      final double visualVelocity = details.velocity.pixelsPerSecond.dx / _width;
-      switch (Directionality.of(context)) {
-      case TextDirection.rtl:
-        _controller.fling(velocity: -visualVelocity);
-        break;
-      case TextDirection.ltr:
-        _controller.fling(velocity: visualVelocity);
-        break;
-    }
-
-    } else if (_controller.value < 0.5) {
-      close();
+    if (widget.type == DrawerType.START) {
```

Comment:
Same comment here.  Rather than duplicating all this code, we can translate `visualVelocity` into `valueVelocity` by first considering the `DrawerAlignment` and then considering the `Directionality`.  Once we do that, we can `fling` the `_controller` using the commuted `valueVelocity`.

Reply:
Fixed